Documentation/ceph-recover-pod-on-LostNode.md

```diff
@@ -0,0 +1,138 @@
+```
+
+**And** then wait for few minutes.
+
+To verify
+
+```console
+kubectl -n rook-ceph get pod
+```
+>```
+> csi-rbd-demo-depl-69cd495f9b-8ldf9                  1/1     Running       0          39m
+>```
+
+And, when you describe the pod now
+
+```console
+$ kubectl -n rook-ceph delete pod csi-rbd-demo-depl-69cd495f9b-nl6hf
+
+Multi-Attach error for volume "pvc-ccb1108e-4d3d-4478-9c16-56a1e3001a7a" Volume is already used by pod(s) csi-rbd-demo-depl-69cd495f9b-rwdxq
+Warning  FailedMount             6m6s               kubelet, ip-10-0-159-129.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[mypvc], unattached volumes=[default-token-s2fdk mypvc]: timed out waiting for the condition
+Normal   SuccessfulAttachVolume  3m6s               attachdetach-controller                AttachVolume.Attach succeeded for volume "pvc-ccb1108e-4d3d-4478-9c16-56a1e3001a7a"
+Warning  FailedMount             94s (x8 over 19m)  kubelet, ip-10-0-159-129.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[mypvc], unattached volumes=[mypvc default-token-s2fdk]: timed out waiting for the condition
+Normal   AddedInterface          78s                multus                                 Add eth0 [10.129.2.21/23]
+Normal   Pulling                 78s                kubelet, ip-10-0-159-129.ec2.internal  Pulling image "docker.io/library/nginx:latest"
+Normal   Pulled                  74s                kubelet, ip-10-0-159-129.ec2.internal  Successfully pulled image "docker.io/library/nginx:latest"
+Normal   Created                 74s                kubelet, ip-10-0-159-129.ec2.internal  Created container web-server
+Normal   Started                 74s                kubelet, ip-10-0-159-129.ec2.internal  Started container web-server
+
+```
+
+# Node lost and still has the device in it
+
+### In this case, you will see pod in `ContainerCreating` state
+
+```console
+kubectl -n rook-ceph get pod
+```
+>```
+> csi-rbd-demo-depl-69cd495f9b-cqbcd           0/1     ContainerCreating   0          24s
+>```
+
+And, when you will describe the pod `Warning` like below
+
+```console
+$ kubectl -n rook-ceph describe pod csi-rbd-demo-depl-69cd495f9b-cqbcd
+
+Warning  FailedMount             115s (x3 over 6m28s)  kubelet, ip-10-0-159-129.ec2.internal  Unable to attach or mount volumes: unmounted volumes=[mypvc], unattached volumes=[mypvc default-token-s2fdk]: timed out waiting for the condition
+Warning  FailedMount             114s (x7 over 7m25s)  kubelet, ip-10-0-159-129.ec2.internal  MountVolume.MountDevice failed for volume "pvc-ccb1108e-4d3d-4478-9c16-56a1e3001a7a" : rpc error: code = Internal desc = rbd image replicapool/csi-vol-1a1bbb34-7461-11eb-a32b-0a580a81020c is still being used
+```
+
+**Solution:**
+
+The solution is to remove the watcher, following the commands below
+
+```console
+$ kubectl -n rook-ceph exec -it rook-ceph-tools-6b4889fdfd-2qrmt -- sh
```

Comment:
```suggestion
kubectl -n rook-ceph exec deploy/rook-ceph-tools -- bash
rbd ls --pool=replicapool
```

Reply:
To map PVC to image in ceph:

```
$ kubectl get pv -o 'custom-columns=NAME:.spec.claimRef.name,PVNAME:.metadata.name,STORAGECLASS:.spec.storageClassName,VOLUMEHANDLE:.spec.csi.volumeHandle'
```
AND using PVC name you want image name for:

```
$ CSIVOL=$(kubectl get pv $(kubectl get pv | grep <pvc_name> | awk '{ print $1 }') -o jsonpath='{.spec.csi.volumeHandle}' | cut -d '-' -f 6- | awk '{print "csi-vol-"$1}')
$ echo $CSIVOL

Reply:
@Madhu-1 @travisn  do we want to add these commands too?